pkg/app/middleware.go

```diff
@@ -64,3 +72,32 @@ func SetupCORSMiddleware(handler http.Handler, domains []string) http.Handler {
 	})
 	return corsHandler.Handler(handler)
 }
+
+func WithIPXEScriptMiddleware(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		// Check plain HTTP requests only
+		if r.TLS == nil {
+			if r.Method != http.MethodGet {
+				http.NotFound(w, r)
+				return
+			}
+			matches := ipxeScriptPattern.FindStringSubmatch(r.URL.Path)
```

Comment:
Do you think we could somehow use https://github.com/openshift/assisted-service/blob/4780c78da601e3cee35a7488ec6a260f9aa16b88/restapi/operations/installer/v2_download_infra_env_files_parameters.go#L51 to save us some of this work?